drivers/usb/usbstor_new/disk.c

```diff
@@ -14,6 +14,72 @@
 #define NDEBUG
 #include <debug.h>
 
+BOOLEAN
+NTAPI
+IsRequestValid(PIRP Irp)
+{
+    ULONG TransferLength;
+    PIO_STACK_LOCATION IoStack;
+    PSCSI_REQUEST_BLOCK Srb;
+
+    DPRINT("IsRequestValid: ... \n");
+
+    IoStack = IoGetCurrentIrpStackLocation(Irp);
+    Srb = IoStack->Parameters.Scsi.Srb;
+
+    if (Srb->SrbFlags & (SRB_FLAGS_DATA_IN | SRB_FLAGS_DATA_OUT))
+    {
+        if ((Srb->SrbFlags & SRB_FLAGS_UNSPECIFIED_DIRECTION) ==
+                             SRB_FLAGS_UNSPECIFIED_DIRECTION)
+        {
+            DPRINT("IsRequestValid: Not valid Srb! Srb->SrbFlags - %p\n", Srb->SrbFlags);
```

Comment:
If these cases are not common, it might make sense to promote to DPRINT1, so it's easier to see what is wrong in the logs.

Reply:
Ok